src/App.tsx

```diff
@@ -1,26 +1,297 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { addTodo, delTodo, get, updTodo } from './api/todos';
+import { TodoInterface } from './types/Todo';
+import { Filter } from './types/filter';
+import { TodoList } from './components/todoList/TodoList';
+import { FilteredTodoList } from './components/footer/FilteredTodoList';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<TodoInterface[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [value, setValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<TodoInterface | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [active, setActive] = useState(false);
+  const [todosForModify, setTodosForModify] = useState<TodoInterface[]>([]);
+
+  const inputForFocusRef = useRef<HTMLInputElement>(null);
+  const notificationRef = useRef<HTMLDivElement>(null);
+
+  const hideNotification = () => {
+    if (notificationRef.current) {
+      notificationRef.current.classList.add('hidden');
+    }
+  };
+
+  const errorHandling = (error: Error) => {
+    if (notificationRef.current) {
+      notificationRef.current.classList.remove('hidden');
+      setErrorMessage(error.message);
+      setTimeout(() => {
+        if (notificationRef.current) {
+          notificationRef.current.classList.add('hidden');
+        }
+      }, 3000);
+    }
+  };
+
+  const newId = () => {
+    const maxId = Math.max(0, ...todos.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+
+  const errorManagement = (er: unknown) => {
+    if (er instanceof Error) {
+      errorHandling(er);
+    }
+  };
+
+  const createTodo = (id: number, title: string): TodoInterface => ({
+    id,
+    userId: 2039,
+    title: title.trim(),
+    completed: false,
+  });
+
+  useEffect(() => {
+    if (inputForFocusRef.current) {
+      inputForFocusRef.current.focus();
+    }
+
+    const fetchTodos = async () => {
+      try {
+        hideNotification();
+
+        const data = await get();
+
+        setTodos(data);
+      } catch (error) {
+        if (error instanceof Error) {
+          errorHandling(error);
+        }
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  const allTodosComplited = useMemo(() => {
+    return todos.every(item => item.completed);
+  }, [todos]);
+
+  useEffect(() => {
+    if (allTodosComplited) {
+      setActive(true);
+    } else {
+      setActive(false);
+    }
+  }, [allTodosComplited, active]);
```

Comment:
```suggestion
  const allTodosComplited = useMemo(() => {
    return todos.every(item => item.completed);
  }, [todos]);
```
it looks redundant, you can just use allTodosComplited var

Reply:
![Знімок екрана 2024-12-13 162017](https://github.com/user-attachments/assets/43e5fede-b376-4c43-b194-b227b7340d8e)